PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:7 + add:0, change matches:
-    _inherit = ['mailing.list']
-    _inherit = ['mailing.mailing']
-    _inherit = ['res.users']
-    _inherit = ['sms.sms']
-    _inherit = ["sms.tracker"]
-    _inherit = ['utm.campaign']
-    _inherit = ['utm.medium']

Total Changes: 15

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mass_mailing_sms/models/mailing_contact.py
+++ b/addons/mass_mailing_sms/models/mailing_contact.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class MailingContact(models.Model):
+    _name = 'mailing.contact'
     _inherit = ['mailing.contact', 'mail.thread.phone']
 
     mobile = fields.Char(string='Mobile')

--- a/addons/mass_mailing_sms/models/mailing_list.py
+++ b/addons/mass_mailing_sms/models/mailing_list.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class MailingList(models.Model):
-    _inherit = ['mailing.list']
+    _inherit = 'mailing.list'
 
     contact_count_sms = fields.Integer(compute="_compute_mailing_list_statistics", string="SMS Contacts")
 

--- a/addons/mass_mailing_sms/models/mailing_mailing.py
+++ b/addons/mass_mailing_sms/models/mailing_mailing.py
@@ -14,7 +14,7 @@ _logger = logging.getLogger(__name__)
 
 
 class MailingMailing(models.Model):
-    _inherit = ['mailing.mailing']
+    _inherit = 'mailing.mailing'
 
     @api.model
     def default_get(self, fields):

--- a/addons/mass_mailing_sms/models/res_users.py
+++ b/addons/mass_mailing_sms/models/res_users.py
@@ -7,7 +7,7 @@ from odoo import api, fields, models, modules, _
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     @api.model
     def _get_activity_groups(self):

--- a/addons/mass_mailing_sms/models/sms_sms.py
+++ b/addons/mass_mailing_sms/models/sms_sms.py
@@ -8,7 +8,7 @@ from odoo.tools.mail import TEXT_URL_REGEX
 
 
 class SmsSms(models.Model):
-    _inherit = ['sms.sms']
+    _inherit = 'sms.sms'
 
     mailing_id = fields.Many2one('mailing.mailing', string='Mass Mailing')
     # Linking to another field than the comodel id allows to use the ORM to create

--- a/addons/mass_mailing_sms/models/sms_tracker.py
+++ b/addons/mass_mailing_sms/models/sms_tracker.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class SmsTracker(models.Model):
-    _inherit = ["sms.tracker"]
+    _inherit = "sms.tracker"
 
     SMS_STATE_TO_TRACE_STATUS = {
         'error': 'error',

--- a/addons/mass_mailing_sms/models/utm.py
+++ b/addons/mass_mailing_sms/models/utm.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class UtmCampaign(models.Model):
-    _inherit = ['utm.campaign']
+    _inherit = 'utm.campaign'
 
     mailing_sms_ids = fields.One2many(
         'mailing.mailing', 'campaign_id',
@@ -78,7 +78,7 @@ class UtmCampaign(models.Model):
 
 
 class UtmMedium(models.Model):
-    _inherit = ['utm.medium']
+    _inherit = 'utm.medium'
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_utm_medium_sms(self):
